src/qtt/algorithms/functions.py

```diff
@@ -215,13 +215,49 @@ def _double_gaussian(x, A_dn, A_up, sigma_dn, sigma_up, mean_dn, mean_up):
     result_dict = {'parameters initial guess': initial_params,
                    'separation': separation, 'split': weigthed_distance_split}
     result_dict['parameters'] = par_fit
+    result_dict['reduced_chi_squared'] = result.redchi
+
     result_dict['left'] = np.take(par_fit, [4, 2, 0])
     result_dict['right'] = np.take(par_fit, [5, 3, 1])
     result_dict['type'] = 'fitted double gaussian'
 
     return par_fit, result_dict
 
 
+def refit_double_gaussian(result_dict, x_data, y_data, gaussian_amplitude_ratio_threshold=8):
+    """ Improve fit of double Gaussian by estimating the initial parameters based on an existing fit
+
+    Args:
+        result_dict(dict): Result dictionary from fit_double_gaussian
+        x_data (array): Independent data
+        y_data (array): Signal data
+        gaussian_amplitude_ratio_threshold (float): If ratio between amplitudes of Gaussian peaks is larger than this fit, re-estimate
+    Returns:
+        Dictionary with improved fitting results
+    """
+    if result_dict['left'][2] > result_dict['right'][2]:
```

Comment:
Instead of all the indexes [0], [1], [2] which actually stands for mean, std and amplitude you can do something like:
```
mean = operator.itemgetter(0)
std = operator.itemgetter(1)
amplitude = operator.itemgetter(2)
```
and then you may do:
```
if amplitude(result_dict['left']) > amplitude(result_dict['right']):
..
gaussian_ratio = amplitude(large_gaussian_parameters) / amplitude(small_gaussian_parameters)
..
        idx = np.logical_and(x_data > mean(large_gaussian_parameters) - 1.5 * std(large_gaussian_parameters),
                             x_data < mean(large_gaussian_parameters) + 1.5 * std(large_gaussian_parameters))
```
which to me is more clear than the hard indexing.

Reply:
That is really cool!